### PR TITLE
Update trusted providers of github actions

### PR DESCRIPTION
**Pull request by liranr23**
Based on
https://github.com/instructlab/dev-docs/blob/main/docs/github-actions-use-policy.md#trusted-providers-of-github-actions, update the trusted providers of instructlab.

Related to https://github.com/instructlab/instructlab/issues/1218

### CODE REVIEW

File: docs/github-actions-providers.md [hunk 1]
```diff
@@ -6,9 +6,11 @@
 * `DavidAnson/markdownlint-cli2-action@*` - Markdownlint-cli2 action
 * `docker/*@*` - Docker actions
 * `pypa/*@*` - Python Packaging actions
-* `rojopolis/spellcheck-github-actions@*` - Spellcheck action
+* `ludeeus/action-shellcheck@*` - Spellcheck action
```
Comment (leseb) [inline docs/github-actions-providers.md:9]:
```suggestion
* `ludeeus/action-shellcheck@*` - Shellcheck action
```

**Review by leseb — CHANGES_REQUESTED** [review]

File: docs/github-actions-providers.md (hunk 1 above)
Comment (liranr23) [reply docs/github-actions-providers.md]: Indeed, done.

**Review by leseb — COMMENTED** [review]
Sorry I missed that during the first review. Can you split your commits like this:

* remove the unused `spellcheck-github-actions`
* add newly supported actions (shellcheck, build-and-inspect-python-package and file-existence-action)

Thanks!

**Review by leseb — APPROVED** [review]
Thanks!

**Comment by liranr23** [comment]
@leseb I can't tell on which `Misspelled words` the spellcheck fails. i guess it's the shellcheck. do you wish me to add it to the dictionary?

**Comment by leseb** [comment]
> @leseb I can't tell on which `Misspelled words` the spellcheck fails. i guess it's the shellcheck. do you wish me to add it to the dictionary?

Yes it is:

```
Misspelled words:
<context> docs/github-actions-providers.md
--------------------------------------------------------------------------------
Shellcheck
```

Please add it to `.spellcheck-en-custom.txt` and run `make spellcheck-sort`.
Thanks!

File: docs/github-actions-providers.md [hunk 2]
```diff
@@ -6,9 +6,11 @@
 * `DavidAnson/markdownlint-cli2-action@*` - Markdownlint-cli2 action
 * `docker/*@*` - Docker actions
 * `pypa/*@*` - Python Packaging actions
-* `rojopolis/spellcheck-github-actions@*` - Spellcheck action
```
[the remark this answers is not in the document]
Comment (liranr23) [reply docs/github-actions-providers.md:9]: I went only with instructlab repo, but e.g here (dev-docs), it is in use. i think now that it should stay(if you confirm it's global)

[the remark this answers is not in the document]
Comment (russellb) [reply docs/github-actions-providers.md:9]: This doc specifies a policy for all instructlab repos

**Review by leseb — APPROVED** [review]
Approving again based on @russellb's suggestion. Thanks!

**Review by bjhargrave — APPROVED** [review]

**Pull request merged**